fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java

```diff
@@ -265,17 +265,16 @@ private void executeDynamicPartition() {
             String tableName;
             boolean skipAddPartition = false;
             OlapTable olapTable;
-            db.readLock();
+            olapTable = (OlapTable) db.getTable(tableId);
```

Comment:
When drop a table from database, a db lock will be held;
Why not get a db read lock here when get a table from database.

Reply:
because when drop table, we need to prevent that other thread drop table or create table too. but for get table, we just need to ensure that the get table operation is thread safe, and the final result is ok, we don't purse the strict consistency here for  better performance.

Reply:
Another consideration is to avoid deadlock， all lock sequence is db lock -> table lock -> other lock, and if we has get table lock and sometimes need to get table again from db, there may cause dead lock, so db get table operation not get db read lock anymore

Reply:
I get, you mean ```Performance is ahead of NPE```.
For the second point about ```lock sequence ```.
I think it should become programming specifications for Doris.
```
1 If try to lock multiple tables or dbs, it should use getXXXInOrderMethod.
2 If try to lock different type locks, lock sequence should be  guaranteed.
3 The appearance of nested locks of the same type should be avoided.
```